.github/workflows/deploy.yaml

```diff
@@ -1,17 +1,43 @@
+# Workflow that builds a Docker image containing the model code,
+# pushes it to the GitHub Container Registry, and then optionally uses
+# that container image to run the model using an AWS Batch job.
+#
+# Images are built on every commit to a PR or main branch in order to ensure
+# that the build continues to work properly, but Batch jobs are gated behind
+# a `deploy` environment that requires manual approval from a codeowner.
+
 name: deploy
 
 on:
   pull_request:
   push:
```

Comment:
**thought:** We actually might _not_ want this to run on pushes. If someone opens a PR with a new feature, they add a bunch of commits, then request a Batch run that succeeds, then there's no reason to re-run the Batch job/deployment once it's pushed into master (except maybe to update the main Docker image?).

I suppose we can just ignore/reject the deployment notice for the merge/push.

Reply:
This makes sense to me! I think we'll indeed want to update the main branch Docker image, but I'll gate the `run-model` job so that it only runs on PRs.

Reply:
Done in 2a85c146466071bfde2d319b2674c074ca2863ff.